src/main/java/cz/cvut/kbss/study/security/CustomSwitchUserFilter.java

```diff
@@ -14,7 +15,7 @@ public class CustomSwitchUserFilter extends SwitchUserFilter {
     @Override
     protected Authentication attemptSwitchUser(HttpServletRequest request) throws AuthenticationException {
         final Authentication switchTo = super.attemptSwitchUser(request);
-        if (switchTo.getAuthorities().stream().anyMatch(a -> SecurityConstants.ROLE_ADMIN.equals(a.getAuthority()))) {
+        if (switchTo.getAuthorities().stream().anyMatch(a -> Role.administrator.name().equals(a.getAuthority()))) {
```

Comment:
Ok, in next step, we will have to go over Role.administrator.name() and replace it with something else (e.g. here it would be rather impersonate role)

Reply:
In this filter, I think it makes sense to use the Administrator role to restrict impersonating other admins

Reply:
@palagdan I was going to remove Role.user and Role.administrator soon so i do not understand.

Anyway now it is ok to have it like like this.

Reply:
Yes, we can change the logic to check if the user has the impersonate role, rather than the admin role.